src/api/auth.ts

```diff
@@ -0,0 +1,143 @@
+import { sha256 } from 'js-sha256';
+import { Buffer } from 'buffer';
+
+interface MyResponseType {
+	status: number;
+	responseText: string;
+}
+
+let userExist = false;
+const clientId = process.env.CLIENT_ID;
+const clientSecret = process.env.CLIENT_SECRET;
+const redirectUri = process.env.REDIRECT_URI;
+
+const scope = `user-read-private user-read-email playlist-read-private playlist-read-collaborative playlist-modify-public playlist-modify-private`;
+
+function generateRandomString(length: number) {
```

Comment:
where are you getting all these functions from? and where are their respective tests?

Reply:
this function is used to generate a state to be passed in the auth flow
the others are used to implement pkce

Reply:
Did you come up with these functions yourself? There's no official package to pull from?

Reply:
No I followed instructions but I will update it to use the npm package "pkce-challenge"

Reply:
Actually I cannot find one that works